imap_processing/spice/pointing_frame.py

```diff
@@ -0,0 +1,282 @@
+    ck_cover = spiceypy.ckcov(
+        str(ck_path), int(id_imap_spacecraft), True, "INTERVAL", 0, "TDB"
+    )
+    num_intervals = spiceypy.wncard(ck_cover)
+    et_start, _ = spiceypy.wnfetd(ck_cover, 0)
+    _, et_end = spiceypy.wnfetd(ck_cover, num_intervals - 1)
+
+    sclk_ticks_start = met_to_sclkticks(repoint_start_met)
+    et_start_repoint = sct_to_et(sclk_ticks_start)
+    sclk_ticks_end = met_to_sclkticks(repoint_end_met)
+    et_end_repoint = sct_to_et(sclk_ticks_end)
+
+    valid_mask = (et_start_repoint >= et_start) & (et_end_repoint <= et_end)
+    et_start_repoint = et_start_repoint[valid_mask]
+    et_end_repoint = et_end_repoint[valid_mask]
+
+    with open_spice_ck_file(pointing_frame_path) as handle:
+        for i in range(len(repoint_start_met)):
+            et_times = _get_et_times(et_start_repoint[i], et_end_repoint[i])
+
+            # Create a rotation matrix
+            rotation_matrix = _create_rotation_matrix(et_times)
+
+            # Convert the rotation matrix to a quaternion.
+            # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.m2q
+            q_avg = spiceypy.m2q(rotation_matrix)
+
+            # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.sce2c
+            # Convert start and end times to SCLK.
+            sclk_begtim = spiceypy.sce2c(int(id_imap_sclk), et_times[0])
+            sclk_endtim = spiceypy.sce2c(int(id_imap_sclk), et_times[-1])
+
+            # Create the pointing frame kernel.
+            # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.ckw02
+            spiceypy.ckw02(
+                # Handle of an open CK file.
+                handle,
+                # Start time of the segment.
+                sclk_begtim,
+                # End time of the segment.
+                sclk_endtim,
+                # Pointing frame ID.
+                int(id_imap_dps),
+                # Reference frame.
+                "ECLIPJ2000",  # Reference frame
+                # Identifier.
+                "IMAP_DPS",
+                # Number of pointing intervals.
+                1,
+                # Start times of individual pointing records within segment.
+                # Since there is only a single record this is equal to sclk_begtim.
+                np.array([sclk_begtim]),
+                # End times of individual pointing records within segment.
+                # Since there is only a single record this is equal to sclk_endtim.
+                np.array([sclk_endtim]),  # Single stop time
+                # Average quaternion.
+                q_avg,
+                # 0.0 Angular rotation terms.
+                np.array([0.0, 0.0, 0.0]),
+                # Rates (seconds per tick) at which the quaternion and
+                # angular velocity change.
+                np.array([1.0]),
+            )
+
+
+def _get_et_times(et_start: float, et_end: float) -> NDArray[np.float64]:
```

Comment:
I'd inline this function into the one place it is used. The naming of this is slightly confusing as it isn't "getting" et_times, you are passing those in and then interpolating between them.